### DISCUSSION TITLE
Strange server list behavior & save files

### DISCUSSION THREAD

**Discussion by emirii** [mbround18/valheim-docker#88] [Help!] [answered]
Hi there, great docker image so far. I just have a few questions I can't seem to answer myself!

My friends and I seem to be having a weird time connecting to the server using the Steam Server List (view > servers> favorites). It seems that one of my friends and I can connect with the IP:2456, but about half my friends can only connect with IP:2457. I also had to restart steam and try to connect a few times for it to finally not show <server not available> and let them actually connect (trying both ports, didn't attempt to try 2458 either though). So I wonder if the ports you connect to are randomly responding? I am not 100% sure on the terminology here.

I tried using a save file from my computer on the docker image, I placed my save files in `./valheim/server/world` but when I used docker-compose up -d it didn't transfer the file. I also transferred the files manually into docker manually but they just got overwritten with a new save. My world name is "Dedicated" as default and I renamed both my save files to Dedicated.db and Dedicated.fwl. I am not sure why anything is not being saved back and fourth from the shared volumes in docker-compose. Maybe that's a docker-compose question though. (my saves and my server folder are both blank on the host)

Basically, I really don't wanna loose my save if my docker image decides to poof!

**Comment by mbround18** [comment, marked as answer]
The connection port should be +1 of the port you set. For example, if your environment variable `PORT` is set to `2456` then the connection port is `2457`.  Would you mind posting your docker-compose file here?

**Reply by emirii** [reply]
Sure! 
```
version: "3"
services:
  valheim:
    image: mbround18/valheim:latest
    ports:
      - 2456:2456/udp
      - 2457:2457/udp
      - 2458:2458/udp
    environment:
      - PORT=2456
      - NAME="Valhalla"
      - WORLD="Dedicated"
      - PASSWORD="boop"
      - TZ=America/Chicago
      - PUBLIC=0
      - AUTO_UPDATE=1
      - AUTO_UPDATE_SCHEDULE="0 7 * * *"
    volumes:
      - ./valheim/saves:/home/docker/valheim/.config/unity3d/IronGate/Valheim
      - ./valheim/server:/home/docker/valheim/valheim
```
I didn't change around much

**Reply by mbround18** [reply]
Well that'll do it :P on your volume mounts they should be `/home/steam` as the base rather than `/home/docker`

**Reply by emirii** [reply]
oh good grief, I thought I was editing the local path. Thanks for that dum dum check!
